src/model.cpp

```diff
@@ -248,28 +279,22 @@ util::optional<cell_size_type> model::local_cell_index(cell_gid_type gid) {
 }
 
 void model::inject_events(const pse_vector& events) {
-    auto& lanes = event_lanes(epoch_.id);
-
-    // Append all events that are to be delivered to local cells to the
-    // appropriate lane. At the same time, keep track of which lanes have been
-    // modified, because the lanes will have to be sorted once all events have
-    // been added.
-    pse_vector local_events;
-    std::set<cell_size_type> modified_lanes;
+    // Push all events that are to be delivered to local cells into the
+    // pending event list for the event's target cell.
     for (auto& e: events) {
         if (e.time<t_) {
-            throw std::runtime_error("model::inject_events(): attempt to inject an event at time " + std::to_string(e.time) + ", when model state is at time " + std::to_string(t_));
+            throw std::runtime_error(
+                "model::inject_events(): attempt to inject an event at time "
+                + std::to_string(e.time)
+                + ", when model state is at time "
```

Comment:
I would add that it should not be before/after: The error displays the state, not what is wrong with it.

Reply:
I have updated the message to indicate that the problem is that the injected event is before the current model time, which is not reasonable/permitted.